### PR TITLE
Bug 1115329 - [Contacts]Device fails to import contacts and go back t…

### PR DESCRIPTION
**Pull request by ferjm**
…o Home automatically, when you import a large number of contacts from memory card. r=francisco

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=61375dd696a182e0520088925c15da250b24bb6f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6fe52d271b5c60393f232ed4b8498aede5beb930

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=93b27e4084c52708877aace6a8ea99479224a8dc

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=232ed341085f5b04698839232bf04c1a1c25bbf5

File: apps/communications/contacts/js/views/settings.js [hunk 1]
```diff
@@ -590,69 +593,81 @@ contacts.Settings = (function() {
...
-              var msg2 = !numDupsMerged ? null : {
-                id: 'contactsMerged',
-                args: {
-                  numDups: numDupsMerged
-                }
-              };
+    function onContacts(text) {
+      return new Promise((resolve, reject) => {
+        if (cancelled) {
+          return;
```
Comment (arcturus) [inline apps/communications/contacts/js/views/settings.js]: nit: reject the promise before return

File: apps/communications/contacts/js/views/settings.js [hunk 2]
```diff
@@ -590,69 +593,81 @@ contacts.Settings = (function() {
...
+        }
 
-              utils.status.show(msg1, msg2);
+        importer = new VCFReader(text);
+        if (!text || !importer) {
+          var error = 'No contacts were found';
+          import_error(error);
+          reject(error);
```
Comment (arcturus) [inline apps/communications/contacts/js/views/settings.js]: nit: return here to finish the execution

File: shared/js/contacts/import/utilities/sdcard.js [hunk 3]
```diff
@@ -139,31 +139,50 @@ if (!utils.sdcard) {
...
+   * @param {Object} file File Object.
+   * @param {Function} cb Function to call for each chunk of vcards retrieved
+   *                      from the file.
    */
-  SdCard.getTextFromFiles = function(fileArray, contents, cb) {
-    contents = contents || '';
-    if (!fileArray || !fileArray.length) {
-      return cb && cb(null, contents);
-    }
+  SdCard.getTextFromFiles = function(file, cb) {
```
Comment (arcturus) [inline shared/js/contacts/import/utilities/sdcard.js]: We should change the function name to express that now we are not passing an array, but a single file:
`getTextFromFile`

File: shared/js/contacts/import/utilities/sdcard.js [hunk 4]
```diff
@@ -139,31 +139,50 @@ if (!utils.sdcard) {
...
+    return new Promise((resolve, reject) => {
+      if (!file) {
```
Comment (arcturus) [inline shared/js/contacts/import/utilities/sdcard.js]: Should we do this check before creating the promise object and stop execution?

Or just `return reject()`

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=43730ac37bb65ded7c5d210eef751d0726007029

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=569a079164f63bda9b4a5a9fc8945c8e4dfe640d

**Pull request merged**